### PR TITLE
Allow Visualiser to be an RStudio addin

### PR DESCRIPTION
**Pull request by Smit-tay**
There appears to be some kind of issue with importing shiny.

### CODE REVIEW

**Comment by Smit-tay** [comment]
Fixes #82

Visualiser works as an add-in.  providing an input UI for various parameters.

**Comment by Smit-tay** [comment]
Fixed the issue which would cause the Visualiser to not actually display, even though it was running.

It seems a Shiny app (which I use for parameter input)  can't launch another Shiny App (the visualiser) trivially.
The work-around, is to make sure the parameter app closes before launching the visualiser.

File: R/addin_handler.R [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+      app_env$closed <- TRUE
+      stopApp()
+
+    })
+  }
+
+  # Run the application and capture the reactive value
+  inputApp <- shiny::shinyApp(ui = ui, server = server)
+  rv <- runApp(inputApp)
```
Comment (RobertASmith) [inline R/addin_handler.R:92]: needs `shiny::` before `runApp`

File: R/addin_handler.R [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      app_env$p_color_with_test = color_with_test
+      app_env$p_color_mod_coverage = color_mod_coverage
+      app_env$p_moderate_coverage_range = input$moderate_coverage_range
+      app_env$p_print_isolated_foo = input$print_isolated_foo
+      app_env$p_show_in_shiny = input$show_in_shiny
+      app_env$p_network_title = input$network_title
+      app_env$p_scale_node_size_by_degree = input$scale_node_size_by_degree
+
+      app_env$closed <- TRUE
+      stopApp()
```
Comment (RobertASmith) [inline R/addin_handler.R:85]: `shiny::` before `stopApp`

File: R/addin_handler.R [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+app_env$closed <- FALSE
+
+# Addin handler
+visualise_project_addin <- function() {
+
+  ui <- shiny::fluidPage(
+    shiny::titlePanel("Visualise Project"),
+    shiny::sidebarLayout(
+      shiny::sidebarPanel(
+        shiny::textInput("project_path", "Project Path", value = ""),
```
Comment (RobertASmith) [inline R/addin_handler.R:29]: Default to value = "." to make it quicker to use when in a project.

File: R/addin_handler.R [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    run_coverage = app_env$p_run_coverage,
+    color_no_test = app_env$p_color_no_test,
+    color_with_test = app_env$p_color_with_test,
+    color_mod_coverage = app_env$p_color_mod_coverage,
+    moderate_coverage_range = app_env$p_moderate_coverage_range,
+    print_isolated_foo = app_env$p_print_isolated_foo,
+    show_in_shiny = app_env$p_show_in_shiny,
+    network_title = app_env$p_network_title,
+    scale_node_size_by_degree = app_env$p_scale_node_size_by_degree)
+
```
Comment (RobertASmith) [inline R/addin_handler.R:114]: Needs a notification of success when the visualiser is produced - maybe a message `message("Visualisation complete, see Viewer pane")` or equivalent.

**Comment by RobertASmith** [comment]
Looks good - I'll make some edits to the UI later to make it prettier, but it seems functional.

**Comment by Smit-tay** [comment]
Done

**Review by RobertASmith — COMMENTED** [review]
This works for me. 

We will want to improve the UI on the modal to:
- reduce the number of options or put them in tabs to keep everything compact
- change the color of the button to make it very obvious what to press. 
- add more information & description. 

However, these edits are mostly aesthetic. The core functionality works.

**Pull request merged**